microprofile/opentracing-smallrye/src/main/java/org/wildfly/extension/microprofile/opentracing/smallrye/TracingDeploymentProcessor.java

```diff
@@ -0,0 +1,150 @@
+package org.wildfly.extension.microprofile.opentracing.smallrye;
+
+import io.opentracing.contrib.jaxrs2.server.SpanFinishingFilter;
+import io.smallrye.opentracing.SmallRyeTracingDynamicFeature;
+import org.jboss.as.ee.weld.WeldDeploymentMarker;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessor;
+import org.jboss.as.server.deployment.Phase;
+import org.jboss.as.web.common.WarMetaData;
+import org.jboss.as.weld.deployment.WeldPortableExtensions;
+import org.jboss.metadata.javaee.spec.ParamValueMetaData;
+import org.jboss.metadata.web.jboss.JBossWebMetaData;
+import org.jboss.metadata.web.spec.FilterMappingMetaData;
+import org.jboss.metadata.web.spec.FilterMetaData;
+import org.jboss.metadata.web.spec.FiltersMetaData;
+import org.jboss.metadata.web.spec.ListenerMetaData;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class TracingDeploymentProcessor implements DeploymentUnitProcessor {
+  public static final Phase PHASE = Phase.POST_MODULE;
+  public static final int PRIORITY = 0x4000;
+
+  @Override
+  public void deploy(DeploymentPhaseContext deploymentPhaseContext) {
+    TracingLogger.ROOT_LOGGER.processingDeployment();
+    DeploymentUnit deploymentUnit = deploymentPhaseContext.getDeploymentUnit();
+
+    configureServiceName(deploymentUnit);
+    addListeners(deploymentUnit);
+    addJaxRsIntegration(deploymentUnit);
+    addCDIExtension(deploymentUnit);
+  }
+
+  @Override
+  public void undeploy(DeploymentUnit deploymentUnit) {
+  }
+
+  private void configureServiceName(DeploymentUnit deploymentUnit) {
+    String serviceName = System.getProperty("JAEGER_SERVICE_NAME");
+    if (null == serviceName || serviceName.isEmpty()) {
+      serviceName = System.getenv("JAEGER_SERVICE_NAME");
+    }
+
+    if (null == serviceName || serviceName.isEmpty()) {
+      serviceName = deploymentUnit.getServiceName().getSimpleName();
+      TracingLogger.ROOT_LOGGER.serviceNameDerivedFromDeploymentUnit(serviceName);
+      System.setProperty("JAEGER_SERVICE_NAME", serviceName);
```

Comment:
What if there are multiple deployments?

Reply:
Great question. I'll change it so that we only *read* properties/env vars.